File: arkhe_multilang/rosetta/agent.cpp
#include <iostream>
#include <string>

struct Agent {
    std::string name;
    int counter = 0;
    Agent(const std::string& n) : name(n) {}
};

void sendMessage(const Agent& sender, Agent& receiver) {
    receiver.counter++;
    std::cout << sender.name << " -> " << receiver.name << ": " << receiver.counter << std::endl;
}

int main() {
    Agent alice("Alice");
    Agent bob("Bob");
    sendMessage(alice, bob);
    sendMessage(alice, bob);
    return 0;
}

File: arkhe_multilang/rosetta/agent.php
<?php

class Agent {
    public $name;
    public $counter = 0;
    function __construct($name) {
        $this->name = $name;
    }
}

function sendMessage($sender, $receiver) {
    $receiver->counter++;
    echo $sender->name . " -> " . $receiver->name . ": " . $receiver->counter . "\n";
}

$alice = new Agent("Alice");
$bob = new Agent("Bob");
sendMessage($alice, $bob);
sendMessage($alice, $bob);
?>
